### PR TITLE
Specify minimum rust version

### LINKED ISSUE #1126: Formally specifcy MSRV
We had a discussion recently about specifying a minimum Rust version. Turns out there actually is a way to put that in the codebase, so users with very old Rust versions will be told in no uncertain terms what the problem is.

### PR DESCRIPTION
**Pull request by slondr**
Resolves #1126

### CODE REVIEW

**Review by robinvd — DISMISSED** [review]

**Comment by eladyn** [comment]
Cool, didn't know that. LGTM! I wonder if we could also pin that version [in CI](https://github.com/Spotifyd/spotifyd/blob/master/.github/workflows/ci.yml) (not CD) so that we notice, when we need to bump the MSRV.

**Comment by slondr** [comment]
Good idea @eladyn. I've added that to this MR.

**Comment by slondr** [comment]
Oh wait, that broke everything.

**Comment by slondr** [comment]
Ok, we're good now.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -56,6 +56,8 @@ jobs:
     strategy:
       matrix:
         os: [macos-latest, ubuntu-latest]
+        toolchain:
+          - "1.62"
```
Comment (eladyn) [inline .github/workflows/ci.yml:60]: That shouldn't be needed, since you aren't using `${{ matrix.toolchain }}` anywhere.

**Comment by Plecra** [comment]
Fantastic, LGTM :) These are almost the exact changes I made in response to #1123 , which this also fixes (and im going to sneakily link to this now ;)) I'm going to make some small changes to CI to ensure we're still getting the most recent lints

**Review by Plecra — APPROVED** [review]
assuming the CI is happy, lets get this done! It'd be nice to have a landing page in the documentation for the error ubuntu users will get:

> ```sh
> error: package `spotifyd v0.3.3 (...)` cannot be built because it requires rustc 1.62 or newer,
> while the currently active rustc version is 1.61.0
> ```

**Comment by Plecra** [comment]
agh, clippy 🙄 fine, will make a quick PR for those

**Review by eladyn — APPROVED** [review]
Thanks for picking this up!

**Pull request merged**